nekoyume/Assets/_Scripts/BlockChain/ActionUnrenderHandler.cs

```diff
@@ -153,67 +156,51 @@ private void ResponseBuy(ActionBase.ActionEvaluation<Buy> eval)
                 return;
             }
 
-            var currentAvatarAddress = States.Instance.CurrentAvatarState.address;
-            var currentAvatarState = eval.OutputStates.GetAvatarState(currentAvatarAddress);
+            var agentAddress = States.Instance.AgentState.address;
+            var avatarAddress = States.Instance.CurrentAvatarState.address;
+            if (!eval.OutputStates.TryGetAvatarStateV2(agentAddress, avatarAddress,
+                out var avatarState))
+            {
+                return;
+            }
 
-            if (eval.Action.buyerAvatarAddress == currentAvatarAddress)
+            var errors = eval.Action.errors.ToList();
+            var purchaseInfos = eval.Action.purchaseInfos;
+            foreach (var purchaseInfo in purchaseInfos)
             {
-                var agentAddress = States.Instance.AgentState.address;
-                var purchaseResults = eval.Action.buyerMultipleResult.purchaseResults;
-                foreach (var purchaseResult in purchaseResults)
+                if (!errors.Exists(tuple => tuple.orderId.Equals(purchaseInfo.OrderId)))
                 {
-                    if (purchaseResult.errorCode == 0)
+                    if (eval.Action.buyerAvatarAddress == avatarAddress) // buyer
```

Comment:
This if statement is better wrapped around the outer foreach statement like below.

```
if (eval.Action.buyerAvatarAddress == avatarAddress) // buyer
{
  foreach (var purchaseInfo in purchaseInfos)
  {
    if (!errors.Exists(tuple => tuple.orderId.Equals(purchaseInfo.OrderId)))
    {
    }
  }
}
else // seller
{
  foreach (var purchaseInfo in purchaseInfos)
  {
    if (!errors.Exists(tuple => tuple.orderId.Equals(purchaseInfo.OrderId)))
    {
    }
  }
}
```

Reply:
fixed👍🏻